.github/workflows/manual-go-test.yml

```diff
@@ -0,0 +1,42 @@
+name: Manual Go Test Run
+
+on:
+  workflow_dispatch:
+
+# When a new revision is pushed to a PR, cancel all in-progress CI runs for that
+# PR. See https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  go-test:
+    name: Go Test (${{ matrix.os }})
+    runs-on: ${{ matrix.run }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            run: ubuntu-22.04
+          - os: darwin
```

Comment:
While running on multiple OS is nice, the darwin / macos costs 10X ([minute multiplier](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions))  expensive so I would like to comment this out as well.

Reply:
The free tier is generous, but once that is exceeded, it's not clear to me who pays on forks, and I don't want any contributor to have a bill or an uncomfortable conversation with their boss.

Reply:
Based on the [Billing page](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions), this repository should count as a "Public repository", which means the MacOS tests won't cost anything, because there's no GH minutes cap.

Reply:
I'm pretty sure forks don't complicate the situation, but I'll respect your concerns on this.

Reply:
I've pushed up a commit disabling the MacOS tests.